src/foam/nanos/mrac/ElectoralServiceServer.js

```diff
@@ -0,0 +1,231 @@
+    }
+  ],
+
+  methods: [
+    {
+      name: 'hasQuorum',
+      type: 'Boolean',
+      args: [
+        {
+          name: 'total',
+          type: 'Integer',
+        }
+      ],
+      javaCode: `
+      return getVotes() >= (total / 2 + 1);
+    `
+    },
+    {
+      name: 'recordResult',
+      synchronized: true,
+      args: [
+        {
+          name: 'result',
+          type: 'Integer',
+        },
+        {
+          name: 'config',
+          type: 'foam.nanos.mrac.ClusterConfig',
+        }
+      ],
+      javaCode: `
+      setVotes(getVotes() + 1);
+      if ( result > getCurrentSeq() ) {
+        setCurrentSeq(result);
+        setWinner(config);
+      }
+    `
+    },
+    {
+      name: 'buildURL',
+      args: [
+        {
+          name: 'config',
+          type: 'foam.nanos.mrac.ClusterConfig'
+        }
+      ],
+      type: 'String',
+      javaCode: `
+      try {
+        // TODO: protocol - http will do for now as we are behind the load balancers.
+        java.net.URI uri = new java.net.URI("http", null, config.getId(), config.getPort(), "/service/electoralService", null, null);
+        return uri.toURL().toString();
+      } catch (java.net.MalformedURLException | java.net.URISyntaxException e) {
+        ((Logger) getX().get("logger")).error(e);
+        return ""; // TODO: 
+      }
+      `
+    },
+    {
+      documentation: `Find the cluster configuration for 'this' (localhost) node.`,
+      name: 'findConfig',
+      type: 'foam.nanos.mrac.ClusterConfig',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        }
+      ],
+      javaCode: `
+      DAO dao = (DAO) x.get("clusterConfigDAO");
+      String hostname = System.getProperty("hostname");
+      if ( SafetyUtil.isEmpty(hostname) ) {
+        hostname = "localhost";
+      }
+      return ((ClusterConfig) dao.find_(x, hostname));
+      `
+    },
+    {
+      name: 'dissolve',
+      javaCode: `
+    ClusterConfig config = findConfig(getX());
+    if ( getState().equals(ElectoralServiceState.IN_SESSION) ) {
+      setElectionTime(new Date());
+      setState(ElectoralServiceState.ELECTION);
+      recordResult(vote(getElectionTime()), config);
+
+      List arr = (ArrayList) ((ArraySink) ((DAO) getX().get("clusterConfigDAO"))
+        .where(
+          AND(
+            AND(
+              EQ(ClusterConfig.REALM, config.getRealm()),
+              EQ(ClusterConfig.REGION, config.getRegion())
+            ),
+            AND(
+              EQ(ClusterConfig.ENABLED, true),
+              EQ(ClusterConfig.STATUS, Status.ONLINE)
+            )
+          )
+        )
+        .select(new ArraySink())).getArray();
+
+      ExecutorService pool = Executors.newFixedThreadPool(arr.size());
+
+      for (int i = 0; i < arr.size(); i++) {
+            ClusterConfig clientConfig = (ClusterConfig) arr.get(i);
+            if ( clientConfig.getId().equals(config.getId())) {
+              break;
```

Comment:
if trying to skip self, then you want 'continue' rather than break.  Continue, will continue the for loop rather than breaking out.
```suggestion
              continue;
```